clerk/side_handler.go

```diff
@@ -90,15 +90,25 @@ func SideHandleMsgEventRecord(ctx sdk.Context, k Keeper, msg types.MsgEventRecor
 	}
 
 	if !bytes.Equal(eventLog.Data, msg.Data) {
-		if !(len(eventLog.Data) > helper.MaxStateSyncSize && bytes.Equal(msg.Data, hmTypes.HexToHexBytes(""))) {
-			k.Logger(ctx).Error(
-				"Data from event does not match with Msg Data",
-				"EventData", hmTypes.BytesToHexBytes(eventLog.Data),
-				"MsgData", hmTypes.BytesToHexBytes(msg.Data),
-			)
-			return hmCommon.ErrorSideTx(k.Codespace(), common.CodeInvalidMsg)
+		if ctx.BlockHeight() > helper.SpanOverrideBlockHeight {
+			if !(len(eventLog.Data) > helper.MaxStateSyncDataLength && bytes.Equal(msg.Data, hmTypes.HexToHexBytes(""))) {
+				k.Logger(ctx).Error(
+					"Data from event does not match with Msg Data",
+					"EventData", hmTypes.BytesToHexBytes(eventLog.Data),
+					"MsgData", hmTypes.BytesToHexBytes(msg.Data),
+				)
+				return hmCommon.ErrorSideTx(k.Codespace(), common.CodeInvalidMsg)
+			}
+		} else {
+			if !(len(eventLog.Data) > helper.MaxStateSyncSize && bytes.Equal(msg.Data, hmTypes.HexToHexBytes(""))) {
```

Comment:
I thought we don't need this in side_hander as well?
cc/ @temaniarpit27 @jdkanani

Reply:
I didn't have that context, Let me know If I should remove it from side_handler